### PR TITLE
Bug Fixes for Toolbox

### PR DESCRIPTION
**Pull request by alschmiedt**
## The basics

- [ ] I branched from develop
- [ ] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves
1. Allows uppercase and lowercase versions of cssConfig and categoryStyle
2. Fixes hovering for categories.

### Proposed Changes
Change the div that handles a click to be the blocklyTreeRow div instead of blocklyToolboxCategory div. The blocklyToolboxCategory div held all of its children, so when there was a nested category and we hovered over the parent category all the children would appear hovered.

### Test Coverage

Tested on:

### CODE REVIEW

File: core/toolbox/category.js [hunk 1]
```diff
@@ -108,15 +115,17 @@ Blockly.ToolboxCategory = function(categoryDef, toolbox, opt_parent) {
...
     'label': 'blocklyTreeLabel',
     'contents': 'blocklyToolboxContents',
     'selected': 'blocklyTreeSelected',
     'openIcon': 'blocklyTreeIconOpen',
-    'closedIcon': 'blocklyTreeIconClosed'
+    'closedIcon': 'blocklyTreeIconClosed',
   };
 
-  Blockly.utils.object.mixin(this.cssConfig_, categoryDef['cssConfig']);
+  var cssConfig = categoryDef['cssConfig'] || categoryDef['cssconfig'];
```
Comment (samelhusseini) [inline core/toolbox/category.js:127]: Curious why this is needed?

File: core/toolbox/category.js [hunk 2]
```diff
@@ -389,7 +415,7 @@ Blockly.ToolboxCategory.prototype.addColourBorder_ = function(colour) {
  * @protected
  */
 Blockly.ToolboxCategory.prototype.getColour_ = function(categoryDef) {
-  var styleName = categoryDef['categorystyle'];
+  var styleName = categoryDef['categorystyle'] || categoryDef['categoryStyle'];
```
Comment (samelhusseini) [inline core/toolbox/category.js:418]: Why is the order of check different here than above? lowercase then uppercase, vs uppercase then lowercase.

Comment (alschmiedt) [reply core/toolbox/category.js:415]: No good reason. Changed them all to be the same.

File: core/toolbox/category.js (hunk 1 above)
Comment (alschmiedt) [reply core/toolbox/category.js]: It's not needed just nice to have. We currently have json and xml for the toolbox definitions and so all of our definitions are lowercase, however I have accidentally done camel case a lot in the JSON when testing and figure developers will as well. So just adding it as an option.

Comment (samelhusseini) [reply core/toolbox/category.js]: Is this something that can be normalized in the xml to json toolbox method? That way we're not checking for both cases all over the place?

Comment (alschmiedt) [reply core/toolbox/category.js]: Not the way it currently is. We only go through the given toolbox definition if it is not the correct format (ie xml, Array<Node>, string). If it is JSON we just return it. I could create a validate method to go through all of the JSON to validate it but I am unsure if this is worth it or not.

Comment (samelhusseini) [reply core/toolbox/category.js]: Ok. Good to keep that option available, but yeah not worth it at this point.

**Pull request merged**